packages/material-ui/src/ListItemText/ListItemText.spec.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import ListItemText from '@material-ui/core/ListItemText';
+
+declare var yada: () => void;
+
+const ListItemTextTest = () => {
+  const FunctionComponent: React.FC<{ prop1: string; prop2: number }> = () => <div />;
+  function notAnElementType(): void {
+    yada();
+  }
+  function SimpleFunctionComponent() {
+    return <div />;
+  }
+  return (
+    <div>
+      <ListItemText />
+      <ListItemText disableTypography />
+      <ListItemText
+        primaryTypographyProps={{ align: 'center' }}
+        secondaryTypographyProps={{ align: 'center' }}
+      />
+      <ListItemText
+        primaryTypographyProps={{ align: 'center', component: 'h1' }}
+        secondaryTypographyProps={{ align: 'center', component: 'p' }}
+      />
+      <ListItemText
+        primaryTypographyProps={{ align: 'center', component: FunctionComponent }}
```

Comment:
@rart I see here you're providing `FunctionComponent` as a value for `component` prop. But you've defined `FunctionComponent` to have two required props, `prop1` and `prop2` which you're not providing. In run time this most probably lead to crush, as `prop1` and `prop2` will be undefined when passed to the `FunctionComponent`. Ans as `FunctionComponent` is developed in TS, it is unlikely that a developer added additional checks for `prop1` and `prop2`, but has relied on TS.

Reply:
Don't see a way of [addressing this at the lower level](https://github.com/mui-org/material-ui/pull/20093#pullrequestreview-374104582) — as requested by eps1lon. Do you?
I think we'd have to do that on an individual nested-typography-declaration component level (i.e. like [in your original PR](https://github.com/mui-org/material-ui/pull/19155)).